### ISSUE TITLE
Database error when trying to perform a backup

### ISSUE THREAD

**Issue by lmscloud-io** [lmscloud-io/moodle-tool_vault#4]
During the "Database modification" pre-check a block_grade_me plugin throws exceptions (mysql):

```
Execute scheduled task: Vault - Site migration (tool_vault\task\cron_task)
... started 19:21:33. Current memory use 62.6MB.
[25/03/24, 19:21:35] [error] [pid 4012167] Operation failed: Coding error detected, it must be fixed by a programmer: Key primary collides with indexmdl_blocgradme_id_ix specified in table block_grade_me
[25/03/24, 19:21:36] [error] [pid 4012167] Operation failed: Coding error detected, it must be fixed by a programmer: Key primary collides with indexmdl_blocgradme_id_ix specified in table block_grade_me
```

**Comment by lmscloud-io** [comment]
This is actually caused by the DB schema not being correct in this block and there is already a PR trying to fix it in the block itslef https://github.com/remotelearner/moodle-block_grade_me/pull/82

However, tool_vault, should not throw exceptions when things like this happen, it's not the only plugin with errors in the DB schema.

**Comment by lmscloud-io** [comment]
This has been fixed in v1.2 (2024032500)
